jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,42 +114,95 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String imagesDir = null;
+	private boolean leaveRunning = false;
+	private boolean shellJob = true;
+	private boolean extUnixSupport = true;
+	private int logLevel = 0;
+	private String logFile = null;
+	private boolean fileLocks = false;
+	private String workDir = null;
+
+	public CRIUSupport setImagesDir(Path imagesDir) throws SecurityException {
+		if (imagesDir != null) {
+			String dir = imagesDir.toAbsolutePath().toString();
+
+			SecurityManager manager = System.getSecurityManager();
+			if (manager != null) {
+				manager.checkPermission(CRIU_DUMP_PERMISSION);
+				manager.checkWrite(dir);
+			}
+			this.imagesDir = dir;
+		}
+		return this;
+	}
+
+	public CRIUSupport setLeaveRunning(boolean leaveRunning) {
+		this.leaveRunning = leaveRunning;
+		return this;
+	}
+
+	public CRIUSupport setShellJob(boolean shellJob) {
+		this.shellJob = shellJob;
+		return this;
+	}
+
+	public CRIUSupport setExtUnixSupport(boolean extUnixSupport) {
+		this.extUnixSupport = extUnixSupport;
+		return this;
+	}
+
+	public CRIUSupport setLogLevel(int logLevel) {
+		if (logLevel > 0) {
+			this.logLevel = logLevel;
+		}
+		return this;
+	}
+
+	public CRIUSupport setLogFile(String logFile) {
+		if (logFile != null && !logFile.contains("/")) {
```

Comment:
I think both `null` and a string containing `File.separatorChar` should trigger `IllegalArgumentException`.